### PR TITLE
Add warnings in the console when the current page is not set

### PR DESCRIPTION
**Pull request by nitriques**
on app init and mediator.notify

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -460,6 +462,11 @@
 				this.init({firstTime: true});
 				this.isInited = true;
 				
+				// Check if we already found it
+				if (!!currentPage) {
```
Comment (PascalPiche) [inline src/app.js]: Dont stop the loop. more page may be needed to be initialized

File: src/app.js [hunk 2]
```diff
@@ -463,6 +465,16 @@
...
+								currentPage: currentPage,
+								previousPage: previousPage,
+								newPage: this
+							}],
+							fx: 'warning'
+						});
+					}
 					// initialise page variable
 					currentPage = this;
 					previousPage = this; // Set the same for the first time
```
Comment (nitriques) [inline src/app.js]: Should we keep previousPage ? 

Like `previousPage = previousPage || this`

Comment (nitriques) [reply src/app.js]: @PascalPiche  ?

File: src/app.js [hunk 3]
```diff
@@ -481,9 +493,13 @@
...
-		notifyAll('app.init', {
-			page: currentPage
-		});
+		if (!currentPage) {
+			App.log({args: 'Can not init application: No current page set.', fx: 'error'});
+		} else {
+			notifyAll('app.init', {
+				page: currentPage
+			});
+		}
```
Comment (nitriques) [inline src/app.js]: Should we still call `notifyAll('app.init')` even if the page was not found ? module should still work, no ?

File: src/app.js (hunk 2 above)
Comment (nitriques) [reply src/app.js]: @PascalPiche Ping!!!

File: src/app.js (hunk 3 above)
Comment (PascalPiche) [reply src/app.js]: go with already notify

File: src/app.js [hunk 4]
```diff
@@ -463,6 +465,16 @@
 				// find if this is our current page
 				// current route found ?
 				if (!!~App.pages._matchRoute(currentRouteUrl, this.routes())) {
+					if (!!currentPage) {
+						App.log({
+							args: ['Previous current page will be changed', {
+								currentPage: currentPage,
+								previousPage: previousPage,
+								newPage: this
```
Comment (PascalPiche) [inline src/app.js]: Should we use newCurrentPage param instead of newPage?

File: src/app.js (hunk 3 above)
Comment (PascalPiche) [reply src/app.js]: But Some module may need fix and may crash with this new edge case. They actually rely on the existence of a page not null.

Comment (nitriques) [reply src/app.js]: Should we give them an empty one ? (null object pattern) ?

Comment (nitriques) [reply src/app.js]: But remember that we always called app.init, even if the page was not found.

File: src/app.js (hunk 4 above)
Comment (nitriques) [reply src/app.js]: Ok

File: src/app.js (hunk 3 above)
Comment (nitriques) [reply src/app.js]: Ok maybe we should create a new handler for this case ?

**Comment by nitriques** [comment]
I've rebased and adressed all fixable concerns.
The only one remaining is the one about the app.init call

**Comment by nitriques** [comment]
@PascalPiche ping !

**Comment by nitriques** [comment]
@PascalPiche Please review 23d275f

**Pull request merged**